gitlabform/processors/project/hooks_processor.py

```diff
@@ -9,23 +13,26 @@ def __init__(self, gitlab: GitLab):
         super().__init__("hooks", gitlab)
 
     def _process_configuration(self, project_and_group: str, configuration: dict):
+        debug("Processing hooks...")
+        project: Project = self.gl.projects.get(project_and_group)
+        hooks_list: List[RESTObject] = project.hooks.list()
```

Comment:
It looks alright to me but I'm not experienced enough. Reading the error message from type check failure here in the CI: https://github.com/gitlabform/gitlabform/actions/runs/6779285076/job/18426101684?pr=635

```
gitlabform/processors/project/hooks_processor.py:18: error: Incompatible types in assignment (expression has type "RESTObjectList | list[RESTObject]", variable has type "list[RESTObject]")  [assignment]
```

Should the type for `hooks_list` be `RESTObjectList` instead?

Reply:
I'll try that! it seemed like it could be either, from the error message, but I don't understand it. Maybe a mistake was to use List instead of list... but then again that's what the docs said.

Reply:
I think I need to explicitly say List[RESTObject] | RESTObjectList ? https://python-gitlab.readthedocs.io/en/stable/api/gitlab.html#gitlab.mixins.ListMixin.list)](https://python-gitlab.readthedocs.io/en/stable/api/gitlab.html#gitlab.mixins.ListMixin.list
I suppose I just thought that an "or", or "|", would accept both, but the return type of list() is what it is...

Reply:
I see you finally figured out the type issue. I didn't know python types could be like that. Learned something new today. Guess we just needed to trust the hint from `mypy`, eh? 😅